### PR TITLE
[Merged by Bors] - feat(Topology/MetricSpace/Pseudo/Defs): add `dense_iff_iUnion_ball` and `dist_eq_of_dist_zero`

### PR DESCRIPTION
**Pull request by pitmonticone**
Upstreamed from the [Carleson](https://github.com/fpvandoorn/carleson) project. 

Co-authored-by: Floris van Doorn

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/MetricSpace/Pseudo/Defs.lean [hunk 1]
```diff
@@ -1127,6 +1131,11 @@ theorem dense_iff {s : Set α} : Dense s ↔ ∀ x, ∀ r > 0, (ball x r ∩ s).
   forall_congr' fun x => by
     simp only [mem_closure_iff, Set.Nonempty, exists_prop, mem_inter_iff, mem_ball', and_comm]
 
+theorem dense_iff_iUnion_ball (s : Set α) : Dense s ↔ ∀ r > 0, ⋃ c ∈ s, ball c r = univ := by
+  simp_rw [eq_univ_iff_forall, mem_iUnion, exists_prop, mem_ball, Dense, mem_closure_iff,
+    forall_comm (α := α)]
+
```
Comment (grunweg) [inline Mathlib/Topology/MetricSpace/Pseudo/Defs.lean:1137]:
```suggestion

**Review by grunweg — COMMENTED** [review]
These lemmas look obvious enough -> let's have a maintainer take a look. Thanks for the PR!
maintainer merge?

File: Mathlib/Topology/MetricSpace/Pseudo/Defs.lean (hunk 1 above)
Comment (pitmonticone) [reply Mathlib/Topology/MetricSpace/Pseudo/Defs.lean:1137]: Done, thanks.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**